packages/web3torrent/src/library/paid-streaming-extension.ts

```diff
@@ -87,15 +89,19 @@ export abstract class PaidStreamingExtension implements Extension {
   }
 
   stop() {
-    this.isForceChoking = true;
-    this.wire.choke();
-    this.executeExtensionCommand(PaidStreamingExtensionNotices.STOP, this.pseChannelId);
+    if (!this.isForceChoking) {
+      this.isForceChoking = true;
+      this.executeExtensionCommand(PaidStreamingExtensionNotices.STOP, this.pseChannelId);
+    }
   }
 
   start() {
-    this.isForceChoking = false;
-    this.wire.unchoke();
-    this.executeExtensionCommand(PaidStreamingExtensionNotices.START);
+    if (this.isForceChoking) {
+      setTimeout(() => {
+        this.isForceChoking = false;
+        this.executeExtensionCommand(PaidStreamingExtensionNotices.START);
+      }, 0);
```

Comment:
Is this a timeout for 0 ms?

Reply:
It is 😕

Reply:
I read that this is a reasonably common hack, equivalent to some small browser-specific timeout. It's basically the minimum timeout possible.

Reply:
I saw this once, might explain it https://github.com/feross/queue-microtask

Reply:
> **Why not use setTimeout(fn, 0)?**
This approach is the most compatible, but it has problems. Modern browsers throttle timers severely, so setTimeout(…, 0) usually takes at least 4ms to run. Furthermore, the throttling gets even worse if the page is backgrounded. If you have many setTimeout calls, then this can severely limit the performance of your program.

Reply:
What's the purpose of the 0ms timeout here?

Reply:
The idea is to let the seeder finish responding all the requests that the leecher made. It dont really like this _"solution"_. I could check the _queue-microtask_ package as a replacement,  or no timeout at all.